### DISCUSSION TITLE
Time Remaining

### DISCUSSION THREAD

**Discussion by spectangulation** [mriscoc/Ender3V2S1#919] [Ideas]
OEM ender firmware seems to calculate time remaining on a print as if the slice it was printing, will be the same slice for every remaining layer, and thus we get wildly different remaining times on the machine screen than in the slicer (where the slicer is the accurate one).

for example: my prints always start with a 15mm/s first later for adhesion, and therefore a 2hr print is shown as 8+ while printing the first layer, and as the speed ramps layer by layer to max speed at layer 3, the time remaining begins to drop.

it would be a small but welcome change to have the time remaining on the screen on my ender by the actual time remaining based on the slicer's duration.

**Comment by mriscoc** [comment]
Use `M73 P## R####` command in your slicer.

**Comment by spectangulation** [comment]
just anywhere in my start code?

**Reply by mriscoc** [reply]
no, in slicer, maybe a plugin. Read Marlin documentation about M73 G-code command.

**Comment by KimmoHop** [comment]
If you have SET_PROGRESS_MANUALLY defined in Configuration_adv.h (e.g. stock Professional firmware bin), 

- PrusaSlicer (and Super Slicer et al) has toggle _Printer Settings -> General -> Supports remaining times_
- Cura has postprocessing plugin _Display Progress On LCD_ which can put out M73 (and M117 to display and M118 to Octoprint?)

Slicers calculate time based on acceleration and jerk given by the user. If printer has different values, no method will be accurate to any time unit. Common to all methods is that they all should end at 0 minutes remaining :)
